src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -62,6 +62,10 @@
   (fn [request]
     (let [environment (get-environment-from-request request)]
       (cond
+        (and (nil? environment)
+             (= "puppet-v3-environment_modules-/*/" (get-in request [:route-info :route-id])))
+        (handler request)
```

Comment:
Ah yeah, I like that idea. I definitely did not want to go with this approach since like you said it bakes in special knowledge in the wrapper function. I'll go ahead with that.